ui/components/app/snaps/snap-ui-renderer/snap-ui-renderer.js

```diff
@@ -30,6 +31,41 @@ const PerformanceTracker = () => {
   return <span data-testid="performance" data-renders={rendersRef.current} />;
 };
 
+const LoadingSpinner = memo(function LoadingSpinner() {
+  return (
+    <Box
+      display={Display.Flex}
+      justifyContent={JustifyContent.center}
+      alignItems={AlignItems.center}
+      height={BlockSize.Full}
+      width={BlockSize.Full}
+    >
+      <PulseLoader />
+    </Box>
+  );
+});
+
+const scrollReducer = (state, action) => {
+  let hasReachedBottom;
+  switch (action.type) {
+    case 'MANUAL_SCROLL':
+      hasReachedBottom = state.hasReachedBottom || action.isAtBottom;
```

Comment:
I don't think this needs to be a mutable variable.

Reply:
yeah true, block scoped it.